examples/hpc/dla-driver/src/lib.rs

```diff
@@ -18,291 +106,501 @@ macro_rules! set_bits {
+            DLA_BUF_STRIDE_Y_BITMASK,
+            reg,
+            y - 1
+        );
+        self.write_u32(DLA_BUF_STRIDE, reg);
+    }
+
+    pub fn set_bias_address(&self, addr: usize) {
+        let reg = set_bits!(
+            DLA_PP_AXI_READ_ADDRESS_OFFSET,
+            DLA_PP_AXI_READ_ADDRESS_BITMASK,
+            0,
+            addr
+        );
+        self.write_u32(DLA_PP_AXI_READ, reg);
+    }
+
+    pub fn get_status(&self) -> u32 {
+        return self.read_u32(DLA_STATUS_ADDR);
+    }
+
+    pub fn set_simd_select(&mut self, mode: SimdBitMode) {
+        let mut reg = self.read_u32(DLA_MAC_CTRL);
+        reg = set_bits!(
+            DLA_SIMD_SELECT_OFFSET,
+            DLA_SIMD_SELECT_BITMASK,
+            reg,
+            mode as usize
+        );
+        self.write_u32(DLA_MAC_CTRL, reg)
+    }
+
+    pub fn get_simd_mode(&self) -> SimdBitMode {
+        let mut reg = self.read_u32(DLA_MAC_CTRL);
+        reg = get_bits!(reg, DLA_SIMD_SELECT_BITMASK);
+        match reg {
+            0 => SimdBitMode::EightBits,
+            1 => SimdBitMode::FourBits,
+            2 => SimdBitMode::TwoBits,
+            _ => SimdBitMode::EightBits,
+        }
+    }
 
-pub fn dla_init() {
-    let mut reg = dla_read_reg(DLA_HANDSHAKE);
-    reg = set_bits!(
-        DLA_HANDSHAKE_BUFFER_ENABLE_OFFSET,
-        DLA_HANDSHAKE_BUFFER_ENABLE_BITMASK,
-        reg,
-        1
-    );
-    reg = set_bits!(
-        DLA_HANDSHAKE_MAC_ENABLE_OFFSET,
-        DLA_HANDSHAKE_MAC_ENABLE_BITMASK,
-        reg,
-        1
-    );
-    reg = set_bits!(
-        DLA_HANDSHAKE_BYPASS_ENABLE_OFFSET,
-        DLA_HANDSHAKE_BYPASS_ENABLE_BITMASK,
-        reg,
-        1
-    );
-    dla_write_reg(DLA_HANDSHAKE, reg);
-
-    dla_set_kernel_size(1, 3, 3);
-    dla_set_input_size(1, 5, 5);
-
-    dla_set_input_data_bank(0);
-    dla_set_kernel_data_bank(8);
-
-    dla_enable_relu(true);
-    dla_enable_bias(true);
-
-    let mut A = [
-        1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3, 4, 4, 4, 4, 4, 5, 5, 5, 5, 5,
-    ];
-    let mut B = [0, 1, 0, 1, 1, 1, 0, 1, 0];
-    dla_write_data_bank(MEMORY_BANK_0_OFFSET, &mut A);
-    dla_write_data_bank(MEMORY_BANK_8_OFFSET, &mut B);
-
-    dla_kernel_data_ready(true);
-    dla_input_data_ready(true);
+    pub fn get_input_bank(&self) -> MemoryBank {
+        let mut reg = self.read_u32(DLA_BUF_DATA_BANK);
+        reg = get_bits!(reg, DLA_BUF_DATA_BANK_B_BITMASK);
+        MemoryBank::from_u32(reg)
+    }
+
+    pub fn get_kernel_bank(&self) -> MemoryBank {
+        let mut reg = self.read_u32(DLA_BUF_DATA_BANK);
+        reg = get_bits!(reg, DLA_BUF_DATA_BANK_A_BITMASK);
+        MemoryBank::from_u32(reg)
+    }
+
+    pub fn get_output_bank(&self) -> MemoryBank {
+        let reg = self.read_u32(DLA_PP_AXI_WRITE);
+        let bank_idx : u32 = (reg - MEMORY_BANK_BASE_ADDR as u32) / MEMORY_BANK_SIZE as u32;
+        MemoryBank::from_u32(bank_idx)
+    }
+
+    pub fn set_mac_clip(&self, clip_amount: usize) {
```

Comment:
Should this be `u32`? `usize` seems unlikely.

Reply:
Mac clip is 5 bits wide, so I don't really know how much does `usize` vs `u32` matter here. Changed to `u32`.

Reply:
`usize` is an architecture dependent type and it immediately raises a question about "why". Better to use a concrete well-defined type.